internal/provider/service_resource.go

```diff
@@ -447,4 +457,13 @@ func serviceToResource(s *tsClient.Service, state serviceResourceModel) serviceR
 		Timeouts:        state.Timeouts,
 		EnableHAReplica: types.BoolValue(s.ReplicaStatus != ""),
 	}
+	if s.VpcEndpoint != nil {
+		if vpcId, err := strconv.ParseInt(s.VpcEndpoint.VpcId, 10, 64); err == nil {
```

Comment:
Could we make a function for that? I think we have the same code in 2 places (could be wrong).
Also I would avoid writing `err == nil`, as usually it's written as `err != nil`, and when skimming through the code it is very easy to misread it. (This is a bit opinionated, but... :sweat_smile:  )

Reply:
I actually wrote it wrong too so probably worth changing.
Sadly this code is not exactly identical, as one mirrors with the API model (having a Spec subbject), and the other one the flattened terraform object